mep/people/views.py

```diff
@@ -7,44 +7,106 @@
 from django.urls import reverse
 from django.utils.html import format_html
 from django.utils.safestring import mark_safe
-from django.views.generic import DetailView, ListView
-from django.views.generic.edit import FormView
+from django.utils.timezone import now
+from django.views.generic import ListView, DetailView
+from django.views.generic.edit import FormView, FormMixin
 from parasolr.django import SolrQuerySet
 
 from mep.accounts.models import Event
 from mep.common.utils import alpha_pagelabels
 from mep.common.views import LabeledPagesMixin
-from mep.people.forms import PersonMergeForm
+from mep.people.forms import PersonMergeForm, MemberSearchForm
 from mep.people.geonames import GeoNamesAPI
 from mep.people.models import Country, Location, Person
 
 
-class MembersList(LabeledPagesMixin, ListView):
+class MembersList(LabeledPagesMixin, ListView, FormMixin):
     '''List page for searching and browsing library members.'''
     model = Person
     template_name = 'people/member_list.html'
     paginate_by = 100
     context_object_name = 'members'
 
+    form_class = MemberSearchForm
+    # cached form instance for current request
+    _form = None
+    #: initial form values
+    initial = {
+        'sort': 'name'
+    }
+
+    def get_form_kwargs(self):
+        kwargs = super().get_form_kwargs()
+        # use GET instead of default POST/PUT for form data
+        form_data = self.request.GET.copy()
+
+        # always use relevance sort for keyword search;
+        # otherwise use default (sort by name)
+        if 'query' in form_data:
+            form_data['sort'] = 'relevance'
+        else:
+            form_data['sort'] = self.initial['sort']
+
+        # use initial values as defaults
+        for key, val in self.initial.items():
+            form_data.setdefault(key, val)
+
+        kwargs['data'] = form_data
+        return kwargs
+
+    def get_form(self, *args, **kwargs):
+        # initialize the form, caching on current instance
+        if not self._form:
+            self._form = super().get_form(*args, **kwargs)
+        return self._form
+
+    #: name query alias field syntax
+    search_name_query = '{!type=edismax qf=$name_qf pf=$name_pf v=$name_query}'
+
+    # map form sort to solr sort field
+    solr_sort = {
+        'relevance': 'score',
+        'name': 'sort_name_sort_s'
+    }
+
     def get_queryset(self):
         sqs = SolrQuerySet().filter(item_type='person') \
-                            .order_by('sort_name_sort_s') \
                             .only(name='name_t', sort_name='sort_name_t',
                                   birth_year='birth_year_i', death_year='death_year_i',
                                   account_start='account_start_i',
                                   account_end='account_end_i',
                                   has_card='has_card_b',
                                   pk='pk_i')
+
         # NOTE: using only / field limit to alias dynamic field names
```

Comment:
Yes, I have an idea about that - was thinking to extend the queryset to configure your default mapping and then filters and other methods could use the mapping.   Could also have a default filter query, so this would be e.g. a `PersonSolrQuerySet` or similar. Not sure where to track that work or when to tackle it, any suggestions?

Reply:
Meant to say: also interested to know more about SQLAlchemy aliases.

Reply:
I think at next iteration planning when we're adding tasks, we might see when to slot this in. Let me think of a user story for parasolr. I can show you an example of SQLAlchemy's tricks.